### PR TITLE
fix(Scripts/HyjalSummit): Rewrite Battle for Mount Hyjal

### PR DESCRIPTION
**Pull request by Gultask**
## Changes Proposed:

This PR proposes changes to:
-  [ ] Core (units, players, creatures, game systems).
-  [X] Scripts (bosses, spell scripts, creature scripts).
-  [X] Database (SAI, creatures, etc).

## Issues Addressed:

- Closes https://github.com/azerothcore/azerothcore-wotlk/issues/4172
- Closes https://github.com/azerothcore/azerothcore-wotlk/issues/4171
- Closes https://github.com/azerothcore/azerothcore-wotlk/issues/4170
- Closes https://github.com/azerothcore/azerothcore-wotlk/issues/17214
- Closes https://github.com/azerothcore/azerothcore-wotlk/issues/17207
- Closes https://github.com/azerothcore/azerothcore-wotlk/issues/17215
- Closes https://github.com/azerothcore/azerothcore-wotlk/issues/17216

## SOURCE:

The changes have been validated through:
- [x] Live research (checked on live servers, e.g Classic WotLK, Retail, etc.)
- [X] Sniffs (remember to share them with the open source community!)
- [ ] Video evidence, knowledge databases or other public sources (e.g forums, Wowhead, etc.)
- [x] The changes promoted by this pull request come partially or entirely from another project (cherry-pick). **Cherry-picks must be committed using the proper --author tag in order to be accepted, thus crediting the original authors, unless otherwise unable to be found**

A lot has been provided by the CMangos team, mainly wave positions.
Pathing has been sniffed by me for the most part.

## Tests Performed:

This PR has been:
- [ ] Tested in-game by the author.
- [ ] Tested in-game by other community members/someone else other than the author/has been live on production servers.
- [ ] This pull request requires further testing and may have edge cases to be tested.

## How to Test the Changes:

- [ ] This pull request can be tested by following the reproduction steps provided in the linked issue
- [ ] This pull request requires further testing. Provide steps to test your changes. If it requires any specific setup e.g multiple players please specify it as well.

1. .ins u all
2. .tele hyjalsummit
3.

## Known Issues and TODO List:

- [x] If summons (totems, spirit wolf, water elementals) are up when a reset happens, the worldserver crashes // It stopped?
- [x] Crash on player death
- [x] General crash prevention
- [x] Thrall, Tyrande
- [x] Trash scripting
- [x] Trash do not move past the first point of their second path (PathEndReached())
- [x] Boss modernization
- [x] DB rework for bases
- [x] Scourge Base
- [x] Archimondo (RP Yell after Azgalor, Spawning in correctly, Spawn after crash/wipe/reset)
- [x] DB Cleanup
- [x] Infernals

## How to Test AzerothCore PRs
 
When a PR is ready to be tested, it will be marked as **[WAITING TO BE TESTED]**.

You can help by testing PRs and writing your feedback here on the PR's page on GitHub. Follow the instructions here:

http://www.azerothcore.org/wiki/How-to-test-a-PR

**REMEMBER**: when testing a PR that changes something **generic** (i.e. a part of code that handles more than one specific thing), the tester should not only check that the PR does its job (e.g. fixing spell XXX) but **especially** check that the PR does not cause any regression (i.e. introducing new bugs).

**For example**: if a PR fixes spell X by changing a part of code that handles spells X, Y, and Z, we should not only test X, but **we should test Y and Z as well**.

### CODE REVIEW

**Comment by heyitsbench** [comment]
Will this resolve the following issues?
https://github.com/azerothcore/azerothcore-wotlk/issues/4170
https://github.com/azerothcore/azerothcore-wotlk/issues/4171
https://github.com/azerothcore/azerothcore-wotlk/issues/4172

**Comment by Gultask** [comment]
some already done others not yet, but yea

**Comment by elthehablo** [comment]
You're insane dude. Nice work

**Comment by heyitsbench** [comment]
> // It stopped?

As temperamental as the weather.

File: data/sql/updates/pending_db_world/rev_1708737709108426300.sql [hunk 1]
```diff
@@ -0,0 +1,2278 @@
...
+(15, 7706, 34158, 0, 0, 13, 0, 5, 3, 2, 1, 0, 0, '', 'Show gossip option if Archimonde is not defeated');
+
+DELETE FROM `creature_template_movement` WHERE (`CreatureId` IN (17906, 17907));
+INSERT INTO `creature_template_movement` (`CreatureId`, `Ground`, `Swim`, `Flight`, `Rooted`, `Chase`, `Random`, `InteractionPauseTimer`) VALUES
+(17906, 1, 0, 1, 0, 0, 0, 0),
+(17907, 1, 0, 1, 0, 0, 0, 0);
+
+UPDATE `creature_text` SET `Emote` = 396 WHERE  `CreatureID` = 17772 AND `GroupID` = 5 AND `ID` = 0;
+
+SET @OGUID := 81488;
```
Comment (Gultask) [inline data/sql/updates/pending_db_world/rev_1708737709108426300.sql:715]: changed
```suggestion
SET @OGUID := 81623;
```

**Comment by Nyeriah** [comment]
Someone did a bad merge and messed up his branch…

**Review by Nyeriah — CHANGES_REQUESTED** [review]
This branch needs to be rebased and force-pushed to the commit before the merge so the polution goes away before the PR can be accepted and merged.

**Comment by Helias** [comment]
> Someone did a bad merge and messed up his branch…

oops

**Pull request closed without merging**